core/src/main/java/google/registry/flows/domain/DomainCreateFlow.java

```diff
@@ -251,6 +252,7 @@ public EppResponse run() throws EppException {
     verifyResourceDoesNotExist(Domain.class, targetId, now, registrarId);
     // Validate that this is actually a legal domain name on a TLD that the registrar has access to.
     InternetDomainName domainName = validateDomainName(command.getDomainName());
+    verifyNotBlockedByBsa(domainName);
```

Comment:
Add an `@error` comment above on the class def.  (Which will require modifying another file that a build test checks, I think `flows.md`.)